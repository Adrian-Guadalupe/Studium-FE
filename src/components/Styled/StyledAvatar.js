import React from 'react';
import { Avatar } from 'antd';
import styled from 'styled-components';
import { ReactComponent as AvatarIcon } from '../../images/ProfileIcons.svg';

const StyledAvatar = ( { avatarUrl, ...props } ) => {
  if( avatarUrl ){
    return ( <StyledAntAvatar src={ avatarUrl } { ...props } size={ 40 }/> );
  }
  return ( <AvatarIcon style={ {
    position: 'absolute', top: '15px', right: '8%',
  } } { ...props }/> );
};

const StyledAntAvatar = styled( Avatar )`
  && {
    position: absolute;
    background-color: #585858;
    filter: contrast(0.5);
    top: 35px;
    right: 8%;
    transform: translate(0, -50%);
  }
`;

StyledAvatar.propTypes = {};

export default StyledAvatar;