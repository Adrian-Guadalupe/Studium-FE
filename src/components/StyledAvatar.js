import React from 'react';
import { Avatar } from 'antd';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledAvatar = props => {
  return ( <StyledAntAvatar icon={ 'user' }>
  
  </StyledAntAvatar> );
};

const StyledAntAvatar = styled( Avatar )`
  && {
    background-color: ${ props => props.theme.darkGray };
  }
`;

StyledAvatar.propTypes = {};

export default StyledAvatar;