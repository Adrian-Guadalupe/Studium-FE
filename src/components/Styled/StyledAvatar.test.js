import React from 'react';
import renderer from 'react-test-renderer';

import StyledAvatar from './StyledAvatar';

test('StyledAvatar render correctly', () => {
    const tree = renderer.create(<StyledAvatar />).toJSON();
    expect(tree).toMatchSnapshot();
});