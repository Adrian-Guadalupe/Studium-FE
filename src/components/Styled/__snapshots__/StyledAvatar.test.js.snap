// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledAvatar render correctly 1`] = `
<svg
  style={
    Object {
      "position": "absolute",
      "right": "8%",
      "top": "15px",
    }
  }
>
  ProfileIcons.svg
</svg>
`;
